app/models/miq_server/role_management.rb

```diff
@@ -50,10 +50,15 @@ def set_active_role_flags
     save
   end
 
-  def set_assigned_roles
+  def sync_assigned_roles
     self.role = ::Settings.server.role
   end
 
+  def ensure_default_roles
+    MiqServer.my_server.add_settings_for_resource(:server => {:role => ENV["DEFAULT_ROLES"]}) if role.blank? && ENV["DEFAULT_ROLES"].present?
```

Comment:
I'm not too concerned about the variable name since I hope it will be short lived, but I namespaced it to avoid conflict.